src/csharp/Grpc.Core/Internal/AsyncCall.cs

```diff
@@ -236,6 +236,7 @@ public void StartServerStreamingCall(TRequest msg)
                     Initialize(details.Channel.CompletionQueue);
 
                     halfcloseRequested = true;
+                    receiveResponseHeadersPending = true;
```

Comment:
`UnsafeSerialize` can throw, right? Looks like `receiveResponseHeadersPending` would never be unset in that case.

Reply:
that's right, but unless callStartedOk = true is set (and it's only set if all the previous steps succeed), the finally statement will run `OnFailedToStartCallLocked();` , which automatically disposes the call. In that case the value of receiveResponseHeadersPending won't matter at all (since the call is already destroyed).